### PR TITLE
require edge condition, set default to on_success, fix #977

### LINKED ISSUE #977: When a new job is created, the default edge will never be run
Note how the edges between a newly created job (bottom-most) or between a downstream job and its parent when the edge hasn't been specifically modified and saved (right-side) is an `∞` symbol. This is invalid and will not let the downstream job run

Until we support the new edge conditions, this needs to go back to being an `on_success` edge.

### PR DESCRIPTION
**Pull request by taylordowns2000**
This fixes #977

## Review checklist

- [x] I have performed a **self-review** of my code
- [x] I have verified that all appropriate **authorization policies** have been implemented and tested
- [x] If needed, I have updated the **changelog**
- [x] Product has **QA'd** this feature

### CODE REVIEW

File: test/lightning/workflows/edge_test.exs [hunk 1]
```diff
@@ -5,10 +5,27 @@ defmodule Lightning.Workflows.EdgeTest do
...
+          condition: :on_job_success
+        })
+
       assert changeset.valid?
     end
 
+    test "edges must have a condition" do
+      changeset =
+        Edge.changeset(%Edge{}, %{workflow_id: Ecto.UUID.generate()})
+        |> IO.inspect()
```
Comment (zacck) [inline test/lightning/workflows/edge_test.exs:20]: Do we need this inspect?

**Review by zacck — COMMENTED** [review]
Looks good to me apart from the Inspect

File: test/lightning/workflows/edge_test.exs (hunk 1 above)
Comment (taylordowns2000) [reply test/lightning/workflows/edge_test.exs]: ack. nope. forgot to remove it, thank you!

**Review by stuartc — APPROVED** [review]
Looks good from a cursory reading through

**Pull request merged**